nlp/berlin/berlin_test.go

```diff
@@ -0,0 +1,184 @@
+				So(*check.LastChecked(), ShouldHappenAfter, timePriorHealthCheck)
+				So(check.LastSuccess(), ShouldBeNil)
+				So(*check.LastFailure(), ShouldHappenAfter, timePriorHealthCheck)
+			})
+
+			Convey("And client.Do should be called once with the expected parameters", func() {
+				doCalls := httpClient.DoCalls()
+				So(doCalls, ShouldHaveLength, 1)
+				So(doCalls[0].Req.URL.Path, ShouldEqual, path)
+			})
+		})
+	})
+
+	Convey("Given a 500 response for health check", t, func() {
+		httpClient := newMockHTTPClient(&http.Response{StatusCode: http.StatusInternalServerError}, nil)
+		searchAPIClient := newBerlinAPIClient(t, httpClient)
+		check := initialTestState
+
+		Convey("When search API client Checker is called", func() {
+			err := searchAPIClient.Checker(ctx, &check)
+			So(err, ShouldBeNil)
+
+			Convey("Then the expected check is returned", func() {
+				So(check.Name(), ShouldEqual, service)
+				So(check.Status(), ShouldEqual, health.StatusCritical)
+				So(check.StatusCode(), ShouldEqual, 500)
+				So(check.Message(), ShouldEqual, service+healthcheck.StatusMessage[health.StatusCritical])
+				So(*check.LastChecked(), ShouldHappenAfter, timePriorHealthCheck)
+				So(check.LastSuccess(), ShouldBeNil)
+				So(*check.LastFailure(), ShouldHappenAfter, timePriorHealthCheck)
+			})
+
+			Convey("And client.Do should be called once with the expected parameters", func() {
+				doCalls := httpClient.DoCalls()
+				So(doCalls, ShouldHaveLength, 1)
+				So(doCalls[0].Req.URL.Path, ShouldEqual, path)
+			})
+		})
+	})
+}
+
+func TestGetBerlin(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	Convey("Given request to find berlin results", t, func() {
+		body, err := json.Marshal(berlinResults)
+		if err != nil {
+			t.Errorf("failed to setup test data, error: %v", err)
+		}
+
+		httpClient := newMockHTTPClient(
+			&http.Response{
+				StatusCode: http.StatusCreated,
+				Body:       io.NopCloser(bytes.NewReader(body)),
+			},
+			nil)
+
+		searchAPIClient := newBerlinAPIClient(t, httpClient)
+
+		Convey("When GetBerlin is called", func() {
+			query := url.Values{}
+			query.Add("q", "census")
+			resp, err := searchAPIClient.GetBerlin(ctx, Options{Query: query})
+
+			Convey("Then the expected response body is returned", func() {
+				So(resp, ShouldNotBeNil)
+				So(resp.Matches[0].Codes[0], ShouldEqual, berlinResults.Matches[0].Codes[0])
+				So(resp.Matches[0].Encoding, ShouldEqual, berlinResults.Matches[0].Encoding)
+				So(resp.Matches[0].Names[0], ShouldEqual, berlinResults.Matches[0].Names[0])
+				So(resp.Matches[0].ID, ShouldEqual, berlinResults.Matches[0].ID)
+				So(resp.Matches[0].Key, ShouldEqual, berlinResults.Matches[0].Key)
+				So(resp.Matches[0].State[0], ShouldEqual, berlinResults.Matches[0].State[0])
+				So(resp.Matches[0].Subdivision[0], ShouldEqual, berlinResults.Matches[0].Subdivision[0])
+				So(resp.Matches[0].Words[0], ShouldEqual, berlinResults.Matches[0].Words[0])
```

Comment:
could just use ShouldResemble here.

Reply:
I know but shouldresemble didn't work for some reason and I got very frustrated and said screw it 
Now I retried it again and it worked I can't really say what I did differently